console/src/components/Services/Data/Add/AddTable.js

```diff
@@ -128,144 +137,245 @@ class AddTable extends Component {
+    const l = cols.length;
+    for (let i = 0; i < l; i++) {
+      if (!this.isValidType(cols[i].type)) {
+        return (
+          'Invalid type (' +
+          cols[i].type.toString() +
+          ') for column ' +
+          i.toString() +
+          ' (' +
+          cols[i].name.toString() +
+          ')'
+        );
+      }
+    }
+    return '';
+  }
+
+  validateNoDupNames(cols) {
+    const l = cols.length;
+    const names = [];
+
+    for (let j = 0; j < l; j++) {
+      names.push(cols[j].name);
+    }
+
+    for (let i = 0; i < l; i++) {
+      const li = names.lastIndexOf(cols[i].name);
+      if (li !== -1 && li !== i) {
+        // slightly tricky code, we just have to find a dup, not all
+        return (
+          'Column ' +
+          i.toString() +
+          ', ' +
+          cols[i].name.toString() +
+          ' is duplicated by column ' +
+          li.toString()
+        );
+      }
+    }
+    return '';
+  }
+
+  /* eslint-disable no-unused-vars */
+  isValidDefault(type, d) {
     return true;
   }
+  /* eslint-enable no-unused-vars */
 
-  submitValidation() {
-    this.props.dispatch(resetValidation());
-    // table name validation
-    if (this.tableNameCheck()) {
-      // column validation.
-      if (this.columnValidation()) {
-        // The primary key validation ensure.
-        if (this.minPrimaryKeyCheck()) {
-          this.props.dispatch(createTableSql());
+  /* punting for now
+    isValidDefault(type, d) {
+        if(d === null || d === "")
+            return true;
+
+        switch (type) {
+            case "integer":
+            case "serial":
+            case "bigint":
+            case "bigserial":
+                return (parseInt(d) !== NaN);
+            case "boolean":
+                return (d === "true" || d === "false");
+            case "numeric":
+            case "float":
+                return (parseFloat(d) !== NaN);
+            case "timestamptz":
+            case "timestamp":
+            case "timetz":
+            case "date":
+                // TODO
+            case "UUID":
+            // TODO
+            case "JSONB":
+            // TODO
+            case "text":
+            default:
+                    return true;
         }
+    }
+*/
+  validateColumnDefaults(cols) {
+    const l = cols.length;
+    for (let i = 0; i < l; i++) {
+      if (!this.isValidDefault(cols[i].type, cols[i].default)) {
+        return (
+          'Invalid default (' +
+          cols[i].default.toString() +
+          ') for column ' +
+          i.toString() +
+          ' (' +
+          cols[i].name.toString() +
+          ')'
+        );
       }
     }
+    return '';
+  }
+
+  // UI has decided we're ready to submit. Validate that the
+  // props have a valid new table, and ask server for new table
+  validateAndSubmit() {
+    this.props.dispatch(resetValidation());
+    const validColumns = this.trimEmptyColumns(this.props.columns);
+
+    if (
```

Comment:
this seems like this may get out of hand quickly with more validations. Create an array of objects with validations and notifications and loop over the array maybe?